lib/marks/link.js

```diff
@@ -24,15 +24,28 @@ export const schema = {
 
 export const serializerSpec = {
   open(state, mark, parent, index) {
-    return "[";
+    state.inEmail = isEmail(mark, parent, index);
+    return state.inEmail ? "<" : "[";
   },
   close(state, mark, parent, index) {
-    return (
-      "](" +
-      mark.attrs.href.replace(/[()"]/g, "\\$&") +
-      (mark.attrs.title ? ` "${mark.attrs.title.replace(/"/g, '\\"')}"` : "") +
-      ")"
-    );
+    const { inEmail } = state;
```

Comment:
I don't really like modifying the serializer state object directly, feels a bit like mucking around with globals, but I know that's how the prosemirror-markdown library does it, so 🤷

Reply:
Yeah, I'm feeling more and more that using a node instead of a mark for links is the correct solution.